File: php/tests/Conformance/AmendFixture/Table/AmendWidget/V0002_AddNote.php
<?php

declare(strict_types=1);

namespace SimpleOrm\Tests\Conformance\AmendFixture\Table\AmendWidget;

use SimpleOrm\Migrations\TableActions;
use SimpleOrm\Migrations\TableMigration;
use SimpleOrm\Tests\Conformance\AmendFixture\AmendWidget;

final class V0002_AddNote extends TableMigration
{
    public function entityClass(): string
    {
        return AmendWidget::class;
    }

    public function action(TableActions $actions): void
    {
        $actions->addColumn('note', 'TEXT');
    }
}

File: go/orm/internal/conformance/amendfixture/Table/AmendWidget/V0002_AddNote.go
package amendwidget

import (
	"github.com/kosmassp/SimpleOrm/go/orm"
	"github.com/kosmassp/SimpleOrm/go/orm/internal/conformance/amendfixture/models"
)

// V0002_AddNote is the draft the model has since moved on from (see AmendWidget's doc comment).
type V0002_AddNote struct {
	orm.TableMigration[models.AmendWidget]
}

func (V0002_AddNote) Action(actions *orm.TableActions) {
	actions.AddColumn("note", "TEXT")
}
